aten/src/ATen/native/Distributions.cpp

```diff
@@ -266,30 +267,75 @@ Tensor& normal_meta_(Tensor& self, double mean, double std, c10::optional<Genera
   return self;
 }
 
+// out tensor float
 Tensor& normal_out(const Tensor& mean, double std, c10::optional<Generator> gen, Tensor& output) {
   return at::native::templates::normal_out_impl<NormalStub, Generator>(output, mean, std, gen);
 }
 
+Tensor& normal_out_meta(const Tensor& mean, double std, c10::optional<Generator> gen, Tensor& output) {
+  CHECK_NORMAL_STD(std);
+  auto std_tensor = at::empty_like(output, MemoryFormat::Contiguous);
+  at::native::templates::resize_output_for_normal(output, mean, std_tensor);
+  return output;
+}
+
+// out float tensor
 Tensor& normal_out(double mean, const Tensor& std, c10::optional<Generator> gen, Tensor& output) {
   return at::native::templates::normal_out_impl<NormalStub, Generator>(output, mean, std, gen);
```

Comment:
What do you think of doing something similar to what we do for the `uniform` meta kernels ([here](https://github.com/pytorch/pytorch/blob/adaf383837238135e93bf456ea4d6a35e7c09f69/aten/src/ATen/native/Distributions.cpp#L248))? In there, we have a (no-op) `UniformMeta` stub that we pass into the template, which lets us re-use all of the error-checking logic in the template for free. Something like:
```
return at::native::templates::normal_out_impl<NormalMeta, Generator>(output, mean, std, gen);
```

That would be kinda nice since it would make each of these functions one-liners (you don't need to construct / resize the meta tensor outputs yourself). But let me know if you think I missed something!

Reply:
yeah, that looks interesting. thx! i'll try to do that when i'm working on this again. i'll let you know if there are any issues

Reply:
Hmm I don't see any changes to the PR. Did you end up going with the `return at::native::templates::normal_out_impl<NormalMeta, Generator>(output, mean, std, gen);` / did the PR get updated?

Reply:
Oh, so this PR adds a version of meta kernels that doesn't use the `NormalMeta` template logic, but a PR further up changes the kernels added in this PR to use that? Okay, let me look more through the stack

Reply:
this is done now